website/blog/2023-11-11-AgentEval/index.mdx

```diff
@@ -0,0 +1,144 @@
+
+---
+title:  AgentEval: Assessing Task Utility for LLM-powered Applications
+authors: julianakiseleva, narabzad
+tags: [LLM, GPT, evaluation, task utility]
+---
+
+![Fig.1: An verification framework](img/agenteval-CQ.png)
+
+**TL;DR:**
+* We introduce a notion of what is task utility in LLM-powered applications
+* We introduce `AgentEval` — the first version of the framework to assess task utility for an arbitrary application. It automaticaly suggests criteria to explain task utility and then quantifies these criteria for logs of your application.
+* We demonstrate the usage of our framework with the [Math Problems dataset](https://microsoft.github.io/autogen/blog/2023/06/28/MathChat)
+
+## Run it yourself
+
+Use the Jupyter notebook for a demonstration of `AgentEval` for [Math Problems dataset](https://microsoft.github.io/autogen/blog/2023/06/28/MathChat) -- [agenteval_cq_math.ipynb](https://github.com/microsoft/autogen/blob/main/notebook/agenteval_cq_math.ipynb)
+
+
+## Introduction
+
+The goal of AutoGen is to make it straightforward and easy to develop LLM-based multi-agent systems for an arbitrary application that helps the end users with their tasks. The next big step forward is what we all want to know: how the developed systems are performing, if they bring utility to our users, and, even more importantly, how we can improve them. Directly evaluating the multi-agent systems is tricky since current approaches mainly rely on success metrics, basically if the agent can accomplish the tasks. Moreover, the definition of success for a particular task is not always given. Introductions of LLMs bring us a lot of abilities that we are eager to convert into actual utilities for end users. In this blog post, we introduce `AgentEval` — the first version of the framework for assessing task utility in LLM-based applications.

```

Comment:
- It would be useful to give an example here of when the success metric is not sufficient. Even better if this example is taken from the Math dataset. For many people task completion success metric == utility, we need to show them a strong example where this is clearly not the case. Similarly, we can give an example for when the definition of success and task completion is not available or not possible to access in real time. 
- Introductions of LLMs bring us a lot of abilities ==> Rapid advances in LLMs and multi-agent systems have introduced a large number of emergent abilities...
- In this blog post, we introduce `AgentEval` — the first version of the framework for assessing task utility in LLM-based applications. ==> We introduce AgentEval — the first version of an evaluation framework to assess task utility for an arbitrary application.

Reply:
@nushib thanks for the comments! Very helpful! Just a quick heads up—we are not writing a scientific paper yet. We are putting together a blog post that should be generally accessible, concise, hopefully a bit entertaining, and not too long.

Reply:
Got it. Yes, including one concrete example would perhaps make it more fun and less researchy.

Reply:
Your feedback is incorporated.